### PR TITLE
[WIP] Update Dropbox to ExternalAccounts

### PR DESCRIPTION
**Pull request by samchrisinger**
# Purpose

To support coming APIv2 addon management, we're updating all of our addon models to utilize the ExternalAccount/Provider architecture. 

# Changes

- Update Dropbox to use ExternalProvider
- Generalize Box *Settings model test cases into generic tests suites
   - implement test suites for Box and Dropbox

# Side Effects
TODO

TODO 
=====
After merge (of parent PR):
* Update GoogleDrive tests to use mixins
* Update Dataverse tests to use mixins
* Create mixins for citation tests
* (Consider) Create/use generic views for citation addons
* (Consider) Create/use generic views for non-standard oauth addons (Dataverse/S3)
* (Consider) Use `pytest.fixture` over mixins for tests.

### CODE REVIEW

File: website/addons/base/__init__.py [hunk 1]
```diff
@@ -764,17 +771,65 @@ class AddonOAuthNodeSettingsBase(AddonNodeSettingsBase):
...
+    def folder_path(self):
+        raise NotImplementedError(
+            "AddonOAuthNodeSettingsBase subclasses must expose a 'folder_path' property."
+        )
+    @property
+    def nodelogger(self):
+        auth = None
+        if self.user_settings:
+            auth = Auth(self.user_settings.owner)
+        return type(
```
Comment (samchrisinger) [inline website/addons/base/__init__.py]: cache the logger

**Comment by samanehsan** [comment]
Please update the dropbox `README.md` with the new redirect uri `http://localhost:5000/oauth/callback/dropbox/`

**Comment by samanehsan** [comment]
When I go to my user settings and click "Connect Account" for dropbox, then click "Cancel" instead of approving dropbox, I get this error message:  
screen shot 2015-11-30 at 9 46 01 am

**Comment by samanehsan** [comment]
When I go to a project and click "Import Account from Profile" for dropbox, my folders get stuck loading: 
screen shot 2015-11-30 at 9 56 02 am

Here are the console errors. The second one happened when I clicked the spinner next to "Full Dropbox":  
screen shot 2015-11-30 at 9 58 32 am

**Pull request closed without merging**